src/cargo/util/errors.rs

```diff
@@ -286,6 +286,20 @@ impl From<Box<CargoError>> for CliError {
 }
 
 // =============================================================================
+// NetworkError trait
+
+pub trait NetworkError: CargoError {
+    fn maybe_spurious(&self) -> bool;
+}
+
+impl NetworkError for git2::Error {
+    fn maybe_spurious(&self) -> bool { true }
+}
+impl NetworkError for curl::ErrCode {
+    fn maybe_spurious(&self) -> bool { true }
```

Comment:
We probably want to inspect the error code here a little more to ensure it actually is spurious